src/test/test-recurse-dir.c

```diff
@@ -129,18 +130,25 @@ int main(int argc, char *argv[]) {
         else
                 p = "/usr/share/man"; /* something hopefully reasonably stable while we run (and limited in size) */
 
+        fd = open(p, O_DIRECTORY|O_CLOEXEC);
+        if (fd < 0 && errno == ENOENT) {
+                log_warning_errno(SYNTHETIC_ERRNO(ENOENT), "Couldn't open directory %s, ignoring: %m", p);
+                return EXIT_TEST_SKIP;
+        }
+        /* If the test directory is on an overlayfs then files and their direcory may return different st_dev
+         * in stat results, which confuses nftw into thinking they're on different filesystems
+         * and don't return the result when the FTW_MOUNT flag is set. */
+        if (fd_is_fs_type(fd, OVERLAYFS_SUPER_MAGIC)) {
+            return EXIT_TEST_SKIP;
+        }
```

Comment:
Please use `log_tests_skipped()` here as well

Reply:
Thanks, serves me right for editing in `vi` in a build VM.

I've added a logging statement rather than removed the braces, because otherwise it skips with no message.